django/db/backends/base/schema.py

```diff
@@ -97,6 +99,7 @@ def execute(self, sql, params=()):
                 "that can't perform a rollback is prohibited."
             )
         # Log the command we're running, then run it
+        sql = force_text(sql)
```

Comment:
the thing is that non-str objects defining a `__str__` method can end up here by introducing `Statement` and returning it from various `_sql()` methods.